### PR TITLE
Add unit-test for Jaeger exporter

### LINKED ISSUE #1142: Add unit-test for Jaeger exporter
As of now, Jaeger exporter only has unit tests for recordable. The unit test for exporter is missing. This needs to be added.

### PR DESCRIPTION
**Pull request by esigo**
Fixes #1142 (issue)

## Changes

Add unit-test for Jaeger exporter

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: exporters/jaeger/CMakeLists.txt [hunk 1]
```diff
@@ -67,4 +67,26 @@ if(BUILD_TESTING)
     TARGET jaeger_recordable_test
     TEST_PREFIX exporter.
     TEST_LIST jaeger_recordable_test)
+
+  add_executable(jaeger_exporter_test test/jaeger_exporter_test.cc)
+  add_definitions(-DGTEST_LINKED_AS_SHARED_LIBRARY=1)
+  if(GMOCK_LIB)
```
Comment (ThomsonTan) [inline exporters/jaeger/CMakeLists.txt:73]: Put this check under MSVC as well?

File: exporters/jaeger/test/jaeger_exporter_test.cc [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+namespace trace      = opentelemetry::trace;
+namespace nostd      = opentelemetry::nostd;
+namespace sdktrace   = opentelemetry::sdk::trace;
+namespace common     = opentelemetry::common;
+namespace sdk_common = opentelemetry::sdk::common;
+
+using namespace jaegertracing;
+using namespace opentelemetry::exporter::jaeger;
+using namespace opentelemetry::sdk::instrumentationlibrary;
+using std::vector;
```
Comment (ThomsonTan) [inline exporters/jaeger/test/jaeger_exporter_test.cc:26]: Is this needed?

**Review by lalitb — APPROVED** [review]

File: exporters/jaeger/test/jaeger_exporter_test.cc [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+class MockThriftSender : public ThriftSender
+{
+public:
+  MOCK_METHOD(int, Append, (std::unique_ptr<JaegerRecordable> &&), (noexcept, override));
+};
+
+TEST_F(JaegerExporterTestPeer, ShutdownTest)
+{
+  auto mock_thrift_ender = new MockThriftSender;
```
[the remark this answers is not in the document]
Comment (esigo) [reply exporters/jaeger/test/jaeger_exporter_test.cc]: thanks, it was a typo.

File: exporters/jaeger/test/jaeger_exporter_test.cc (hunk 2 above)
Comment (esigo) [reply exporters/jaeger/test/jaeger_exporter_test.cc]: thanks, removed

File: exporters/jaeger/CMakeLists.txt (hunk 1 above)
Comment (esigo) [reply exporters/jaeger/CMakeLists.txt]: thanks, done

**Review by ThomsonTan — APPROVED** [review]

**Pull request merged**